### ISSUE TITLE
Add the ability to manage deployments from the UI

### ISSUE THREAD

**Issue by kneal** [go-vela/community#111]
## Description

A user should be able to trigger a deployment and view a history of deployments from the ui
## Value

This allows user to manage their deployments without downloading the CLI or interacting with the API

## Definition of Done

Deployments are manageable within the UI

## Impacted Personas (Optional)

Vela users using the deployment evenet

**Comment by JayCeeJr** [comment]
I would like to take on this issue, Here is a mockup of the UI, any feedback?
![Deployment-Button](https://user-images.githubusercontent.com/1265665/122823243-b9e20900-d2a4-11eb-855e-fda8a1ded203.png)
![Deployment-Form](https://user-images.githubusercontent.com/1265665/122823242-b9497280-d2a4-11eb-83d7-447c27d8d070.png)

**Comment by wass3r** [comment]
Thanks for the interest and taking a stab with the mock(s)!

With regard to the first mock: I think there was some discussion elsewhere around this idea already, but I think this could take on the same approach as "Secrets" and occupy a slot on the tab next to "Builds" from where you can add new deployments.

The view would list all existing deployments with information similar to what you see with `vela get deployment ... --output wide` via the CLI. Ideally, you would also be able to get the associated "Build" so you can link to it from that view. This might entail some changes on the server side?

For the second screenshot: i think this more or less mimics adding a secret with different fields as you are doing. We could probably turn "Reference" and "Task" into input fields (vs textareas) - I don't envision those to contain large amounts of text. With the comments on the first screenshot, we'd get the highlighted tab to indicate that we are on deployments so we wouldn't need the "breadcrumb" addition.

Let me know what your thoughts.

**Comment by JordanSussman** [comment]
> The view would list all existing deployments with information similar to what you see with vela get deployment ... --output wide via the CLI. Ideally, you would also be able to get the associated "Build" so you can link to it from that view. This might entail some changes on the server side?

Would you think about removing the deployment selector within the builds tab? Seems like some of the data would be duplicated between the two tabs.

**Comment by wass3r** [comment]
Not necessarily though that did cross my mind. I was just thinking about being able to link to a build that's associated with a deployment. Currently, `vela get deployment ...` and it's associated Vela API endpoints are basically just proxies to the respective GitHub endpoints. So, in some ways this would just be Vela's take on `<github-url>/<org>/<repo>/deployments` (eg. https://github.com/go-vela/docs/deployments) with the addition of being able to add a deployment and a link to view the associated build. Might be valuable to show off some of the metadata pertaining to each deployment (such as payload data) in the "Deployments" tab. Since deployments aren't currently stored in the Vela database, the build association might be a bit more involved if it were to be part of the requirement.

TL;DR: I think they can live on as separate sections.

**Comment by JayCeeJr** [comment]
Would like some feedback on the Draft PR. https://github.com/go-vela/ui/pull/414

I added the following:
 - Form for triggering a deployment.
 - Deployment tab to the repo view.

I think after triggering a deployment we could send the user back to the builds page since the deployment should be the top build in the list.

I was also thinking about showing some of the deployment metadata in the builds page and possibly adding a "Promote Deployment" button which could fill the deployment form from an existing deployment. But that could be another story.
